source/Command.cpp

```diff
@@ -272,45 +272,49 @@ void Command::Load(const DataNode &node)
 // Reset this to an empty command.
 void Command::Clear()
 {
-	*this = Command();
+	state = 0;
+	weapon.clear();
+	aim.clear();
 }
 
 
 
 // Clear any commands that are set in the given command.
-void Command::Clear(Command command)
+void Command::Clear(const Command &command)
 {
 	state &= ~command.state;
 }
 
 
 
 // Set any commands that are set in the given command.
-void Command::Set(Command command)
+void Command::Set(const Command &command)
 {
 	state |= command.state;
 }
 
 
 
 // Check if any of the given command's bits that are set, are also set here.
-bool Command::Has(Command command) const
+bool Command::Has(const Command &command) const
 {
-	return (state & command.state);
+	if(state & command.state)
+		return true;
+	return weapon.intersects(command.weapon);
 }
 
 
 
 // Get the commands that are set in this and in the given command.
-Command Command::And(Command command) const
+Command Command::And(const Command &command) const
 {
 	return Command(state & command.state);
 }
 
 
 
 // Get the commands that are set in this and not in the given command.
-Command Command::AndNot(Command command) const
+Command Command::AndNot(const Command &command) const
 {
 	return Command(state & ~command.state);
```

Comment:
I guess this originally also included the fire bits, but for this function it feels strange to "add the not".